var/spack/repos/builtin/packages/abacus/package.py

```diff
@@ -6,40 +6,214 @@
+    )
+    version(
+        "2.3.1",
+        sha256="e9522469f499bcb1c57ec05e9866403e4142c97762d7662b4b638f7493bf5122",
+        deprecated=True,
+    )
+    version(
+        "2.3.0",
+        sha256="c5a803a1a596983681d65aff46762f2f6b82b5f50449a53d0ec7dad368d35842",
+        deprecated=True,
+    )
+    version(
+        "2.2.4",
+        sha256="1f3923d32c392fd2564d137b076134aebdc461dc3148c535a00f9b10a615157a",
+        deprecated=True,
+    )
+    version(
+        "2.2.3",
+        sha256="88dbf6a3bdd907df3e097637ec8e51fde13e2f5e0b44f3667443195481320edf",
+        deprecated=True,
+    )
+    version(
+        "2.2.2",
+        sha256="4a7cf2ec6e43dd5c53d5f877a941367074f4714d93c1977a719782957916169e",
+        deprecated=True,
+    )
+    version(
+        "2.2.1",
+        sha256="14feca1d8d1ce025d3f263b85ebfbebc1a1efff704b6490e95b07603c55c1d63",
+        deprecated=True,
+    )
+    version(
+        "2.2.0",
+        sha256="09d4a2508d903121d29813a85791eeb3a905acbe1c5664b8a88903f8eda64b8f",
+        deprecated=True,
+    )
 
+    variant("mpi", default=True, description="Enable MPI support")
     variant("openmp", default=True, description="Enable OpenMP support")
+    variant(
+        "lcao",
+        default=True,
+        description="Enable Linear Combinition of Atomic Orbital calculation",
+        when="+mpi",
+    )
+    variant("libxc", default=True, description="Support additional functionals via libxc")
+    variant(
+        "elpa",
+        default=True,
+        description="Enable optimised diagonalisation routines from ELPA",
+        when="+lcao",
+    )
+    variant("mathlib", default=False, description="Enable ABACUS's builtin libm")
+    variant(
+        "tests", default=False, description="Build ABACUS unit tests", when="build_system=cmake"
+    )
+    variant(
+        "benchmarks",
+        default=False,
+        description="Enable ABACUS's builtin benchmark tests",
+        when="+tests",
+    )
+    variant("rocm", default=False, description="(Experimental)Enable rocm support")
+    # TODO: Add support for
+    # LibRI(https://github.com/abacusmodeling/LibRI),
+    # LibComm(https://github.com/abacusmodeling/LibComm),
+    # Libnpy(https://github.com/llohse/libnpy/),
+    # DeePKS(https://github.com/deepmodeling/deepks-kit),
+    # DeePMD(https://github.com/deepmodeling/deepmd-kit),
+    # LibPAW-interface(https://github.com/wenfei-li/libpaw_interface),
+    # At 2024-1-30, none of above have a spack package.
+
+    depends_on("fftw-api@3")
+    depends_on("blas")
+    depends_on("lapack")
+
+    with when("+mpi"):
+        depends_on("mpi", type=("build", "link", "run"))
+    with when("+libxc"):
+        depends_on("libxc")
+    with when("+lcao"):
+        depends_on("cereal")
+        depends_on("scalapack")
+    with when("+elpa"):
+        depends_on("elpa")
+    with when("+tests"):
+        depends_on("googletest")
+    with when("+benchmarks"):
+        depends_on("benchmark")
```

Comment:
I guess these are `type="test"`, even though we have a few long-standing issues with modeling test deps. :thinking:

Reply:
Yes. Spack has a dependent type `test` in addition to `build, link, run`?

Reply:
solved by 3c96d82.